### PR TITLE
Adjust MoneriumPack to v2.7.2 of @monerium/sdk

### PR DESCRIPTION
**Pull request by einaralex**
Adjust the MoneriumPack to v2.7.2 of @monerium/sdk.

The latest version encapsulates all management of access tokens, order web sockets and more.

Added session storage for the selected safe to the example so the same address would be selected after returning from the Monerium auth flow.

### CODE REVIEW

**Comment by einaralex** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by dasanra** [comment]
Hey @einaralex , thank you for your PR.

We were focused on adding ethers v6 compatibility lately so we couldn't check this before. I see there may be conflicts between current main branch and your PR. Could you check and resolve those?

Thank you!

**Comment by einaralex** [comment]
@dasanra Done :) Thank you 🙏

File: packages/onramp-kit/package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "access": "public"
   },
   "dependencies": {
-    "@monerium/sdk": "2.6.5",
+    "@monerium/sdk": "2.8.3",
```
Comment (dasanra) [inline packages/onramp-kit/package.json:38]:
```suggestion
    "@monerium/sdk": "^2.8.3",
```

File: packages/onramp-kit/src/packs/monerium/SafeMoneriumClient.ts [hunk 2]
```diff
@@ -31,8 +40,8 @@ export class SafeMoneriumClient extends MoneriumClient {
    * @param environment The Monerium environment
```
Comment (dasanra) [inline packages/onramp-kit/src/packs/monerium/SafeMoneriumClient.ts:40]:
```suggestion
   * @param moneriumOptions The Monerium options object
```

File: packages/onramp-kit/src/packs/monerium/SafeMoneriumClient.ts [hunk 3]
```diff
@@ -154,49 +163,32 @@ export class SafeMoneriumClient extends MoneriumClient {
...
+    return Number(BigInt(await this.#safeSdk.getChainId()))
+  }
+
   /**
    * Get the corresponding Monerium SDK Chain from the current chain id
    * @returns The Chain
    */
   async getChain(): Promise<Chain> {
-    const chainId = await this.#safeSdk.getChainId()
+    const chainId = BigInt(await this.#safeSdk.getChainId())
```
Comment (dasanra) [inline packages/onramp-kit/src/packs/monerium/SafeMoneriumClient.ts:179]: It already returns a BigInt
```suggestion
    const chainId = await this.#safeSdk.getChainId()
```

File: packages/onramp-kit/src/packs/monerium/SafeMoneriumClient.ts [hunk 4]
```diff
@@ -154,49 +163,32 @@ export class SafeMoneriumClient extends MoneriumClient {
...
+    return getMoneriumChain(Number(chainId))
   }
 
   /**
    * Get the corresponding Monerium SDK Network from the current chain id
    * @returns The Network
    */
   async getNetwork(): Promise<Networks> {
-    const chainId = await this.#safeSdk.getChainId()
+    const chainId = BigInt(await this.#safeSdk.getChainId())
```
Comment (dasanra) [inline packages/onramp-kit/src/packs/monerium/SafeMoneriumClient.ts:189]: It already returns a BigInt

```suggestion
    const chainId = await this.#safeSdk.getChainId()
```

File: packages/onramp-kit/src/packs/monerium/MoneriumPack.ts [hunk 5]
```diff
@@ -45,100 +35,53 @@ export class MoneriumPack extends OnRampKitBasePack {
...
+      },
+      options.safeSdk
+    )
   }
 
   /**
    * This method initialize the flow with Monerium in order to gain access to the resources
    * using the access_token. Return a initialized {@link SafeMoneriumClient}
-   * @param options The MoneriumOpenOptions object
    * @returns A {@link SafeMoneriumClient} instance
```
Comment (dasanra) [inline packages/onramp-kit/src/packs/monerium/MoneriumPack.ts:51]:
```suggestion
   * @param {MoneriumOpenOptions} [options] The MoneriumOpenOptions object
   * @returns A {@link SafeMoneriumClient} instance
```

File: packages/onramp-kit/src/packs/monerium/MoneriumPack.ts [hunk 6]
```diff
@@ -156,18 +100,11 @@ export class MoneriumPack extends OnRampKitBasePack {
...
 
-    const authFlowUrl = this.client.getAuthFlowURI({
-      client_id: this.#config.clientId,
-      redirect_uri: redirectUrl,
+    await this.client.authorize({
       address: safeAddress,
       signature: '0x',
-      chain: await this.client.getChain(),
-      network: await this.client.getNetwork()
+      chainId: await this.client.getChainId()
```
Comment (dasanra) [inline packages/onramp-kit/src/packs/monerium/MoneriumPack.ts:106]: I would suggest that you align this in your SDK. For some methods, like `linkAddress`, you use `chain` and `network`. This one now uses directly `chainId`.

File: packages/onramp-kit/src/packs/monerium/SafeMoneriumClient.ts [hunk 7]
```diff
@@ -154,26 +163,22 @@ export class SafeMoneriumClient extends MoneriumClient {
     }
   }
 
+  /**
+   * Get the current chain id
+   * @returns The Chain Id
+   */
+  async getChainId(): Promise<number> {
+    return Number(BigInt(await this.#safeSdk.getChainId()))
```
Comment (dasanra) [inline packages/onramp-kit/src/packs/monerium/SafeMoneriumClient.ts:171]:
```suggestion
    return Number(await this.#safeSdk.getChainId())
```

**Comment by yagopv** [comment]
I tested the integration and is working nicely

**Review by yagopv — APPROVED** [review]

**Pull request merged**